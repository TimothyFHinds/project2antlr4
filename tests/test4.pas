program test4;

var 
num1 : REAL;
num2 : REAL;



begin 
    num1 := 10;
    
    while num1 < 20 do
    begin
        writeln('value of num1');
        writeln(num1);
        num1 := num1 + 1;

        if(num1 > 15) then
            break;
    end;
    writeln();

    for num2 := 10 to 20 do 
    begin 
        writeln('value of num2: ');
        writeln(num2);

        if(num2 > 14) then
            break;
    end;

    writeln();

    num1 := 10;
    
    
    while num1 < 20 do
    begin
        writeln('value of num1');
        writeln(num1);
        num1 := num1 + 1;

        if(num1 > 15) then
            break;
    end;
    
    {* can have multiple loops after each other*}
    writeln('expect output 15 for each loop');    

end.