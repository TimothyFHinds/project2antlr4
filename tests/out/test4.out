'value of num1'
10.0
'value of num1'
11.0
'value of num1'
12.0
'value of num1'
13.0
'value of num1'
14.0
'value of num1'
15.0

'value of num2: '
10.0
'value of num2: '
11.0
'value of num2: '
12.0
'value of num2: '
13.0
'value of num2: '
14.0
'value of num2: '
15.0

'value of num1'
10.0
'value of num1'
11.0
'value of num1'
12.0
'value of num1'
13.0
'value of num1'
14.0
'value of num1'
15.0
'expect output 15 for each loop'
